### ISSUE TITLE
Saving .png figure failing on OSX backend

### ISSUE THREAD

**Issue by dstansby** [matplotlib/matplotlib#9156]
### Bug report
Saving `.png` files is currently failing on the osx backend. I'm guessing this is due to #6603 (see the error message below)

**Code for reproduction**

```python
import matplotlib
matplotlib.use('MacOSX')
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
fig.savefig('test.png')
```

**Actual outcome**

```python
  File "test.py", line 6, in <module>
    fig.savefig('test.png')
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/figure.py", line 1813, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backend_bases.py", line 2258, in print_figure
    **kwargs)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_agg.py", line 507, in print_png
    FigureCanvasAgg.draw(self)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_agg.py", line 429, in draw
    toolbar.set_cursor(cursors.WAIT)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_macosx.py", line 188, in set_cursor
    _macosx.set_cursor(cursor)
SystemError: <built-in function set_cursor> returned NULL without setting an error
```

**Expected outcome**

**Matplotlib version**

  * Operating System: OSX
  * Matplotlib Version: Current master branch installed using pip
  * Python Version: 3.6.2

**Comment by jenshnielsen** [comment]
I can reproduce this with 2.1rc1 Agreed that this is release critical

**Comment by anntzer** [comment]
The offending section is in `_macosx.m` at
```
static PyObject*
set_cursor(PyObject* unused, PyObject* args)
{
    int i;
    if(!PyArg_ParseTuple(args, "i", &i)) return NULL;
    switch (i)
    { case 0: [[NSCursor pointingHandCursor] set]; break;
      case 1: [[NSCursor arrowCursor] set]; break;
      case 2: [[NSCursor crosshairCursor] set]; break;
      case 3: [[NSCursor openHandCursor] set]; break;
      default: return NULL;
    }
    Py_INCREF(Py_None);
    return Py_None;
}
```
which just seems to be poorly written code (indeed you need to set an exception if you're going to return NULL, see https://docs.python.org/3.6/c-api/exceptions.html#c.PyErr_SetString).  In any case, cursors.WAIT is equal to 4 (see backend_tools.py) and it would seem to one only needs to add a case to the switch, but...
https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/CursorMgmt/Tasks/ChangingCursors.html

> There is no NSCursor instance for the wait cursor, because the system automatically displays it at the appropriate times.

I guess if OSX knows better than you then we should just do nothing in that case (`case 4: break;`).

**Comment by dstansby** [comment]
Thanks @anntzer !

**Comment by tacaswell** [comment]
closed by #9157
